### PR TITLE
show PDF and PPt in dropdownlist

### PR DESCRIPTION
**Pull request by Tahia50**

### CODE REVIEW

File: FileReaderApp/DOPDropDownMenu.m [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+@property (nonatomic, strong) UITableView *tableView;
+//data source
+@property (nonatomic, copy) NSArray *array;
+//layers array
+@property (nonatomic, copy) NSArray *titles;
+@property (nonatomic, copy) NSArray *indicators;
+@property (nonatomic, copy) NSArray *bgLayers;
+@end
+
+
```
Comment (jannatulabeda) [inline FileReaderApp/DOPDropDownMenu.m]: Remove extra line

File: FileReaderApp/DOPDropDownMenu.m [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+- (void)setDataSource:(id<DOPDropDownMenuDataSource>)dataSource {
+    _dataSource = dataSource;
+    
+    //configure view
+    if ([_dataSource respondsToSelector:@selector(numberOfColumnsInMenu:)]) {
+        _numOfMenu = [_dataSource numberOfColumnsInMenu:self];
+    } else {
+        _numOfMenu = 1;
+    }
+    //CGFloat textLayerInterval = self.frame.size.width / ( _numOfMenu * 2);
```
Comment (jannatulabeda) [inline FileReaderApp/DOPDropDownMenu.m]: Remove commented code

File: FileReaderApp/ViewController.m [hunk 3]
```diff
@@ -7,21 +7,62 @@
...
+    UITouch *touch = [[event allTouches] anyObject];
+    if ([touch view] != self.menu) {
+        [self.menu dismiss];
+    }
+    [super touchesBegan:touches withEvent:event];
+}
+
+- (void)setUpDropDownMenu {
+    CGSize screenSize = [UIScreen mainScreen].bounds.size;
+    DOPDropDownMenu *menu = [[DOPDropDownMenu alloc] initWithOrigin:CGPointMake((screenSize.width - 250.f) / 2.f, 100.f) andHeight:40.f];
```
Comment (jannatulabeda) [inline FileReaderApp/ViewController.m]: Don't use hard coded value

File: FileReaderApp/DOPDropDownMenu.h [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+//
+//  Created by weizhou on 9/26/14.
+//  Copyright (c) 2014 fengweizhou. All rights reserved.
+//
+
+#import <UIKit/UIKit.h>
+
+@interface DOPIndexPath : NSObject
+
+@property (nonatomic, assign) NSInteger column;
```
Comment (hasanmainul) [inline FileReaderApp/DOPDropDownMenu.h]: Convention. Storage then atomicity

File: FileReaderApp/DOPDropDownMenu.h [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+@property (nonatomic, assign) NSInteger row;
```
Comment (hasanmainul) [inline FileReaderApp/DOPDropDownMenu.h]: Same as above

File: FileReaderApp/ViewController.m [hunk 6]
```diff
@@ -7,21 +7,64 @@
...
+    UITouch *touch = [[event allTouches] anyObject];
+    if ([touch view] != self.menu) {
+        [self.menu dismiss];
+    }
+    [super touchesBegan:touches withEvent:event];
+}
+
+- (void)setUpDropDownMenu {
+    CGSize screenSize = [UIScreen mainScreen].bounds.size;
+    DOPDropDownMenu *menu = [[DOPDropDownMenu alloc] initWithOrigin:CGPointMake((screenSize.width - kDropDownWidth) / 2.f, 100.f) andHeight:40.f];
```
Comment (hasanmainul) [inline FileReaderApp/ViewController.m]: if the values are Constants, please maintain macro

File: FileReaderApp/ViewController.m [hunk 7]
```diff
@@ -7,21 +7,64 @@
...
 
 @end
 
 @implementation ViewController
 
 - (void)viewDidLoad {
     [super viewDidLoad];
-    // Do any additional setup after loading the view, typically from a nib.
+    self.navigationItem.title = @"FileReader";
+    NSString *filePath = [[[NSBundle mainBundle] resourcePath] stringByAppendingPathComponent:@"FilesFolder"];
```
Comment (jannatulabeda) [inline FileReaderApp/ViewController.m]: Define the folder name in macro please

**Comment by dina28** [comment]
ok 👍

**Comment by hasanmainul** [comment]
@Tahia50 👍

File: FileReaderApp/ViewController.m [hunk 8]
```diff
@@ -7,21 +7,67 @@
...
+    UITouch *touch = [[event allTouches] anyObject];
+    if ([touch view] != self.menu) {
+        [self.menu dismiss];
+    }
+    [super touchesBegan:touches withEvent:event];
+}
+
+- (void)setUpDropDownMenu {
+    CGSize screenSize = [UIScreen mainScreen].bounds.size;
+    DOPDropDownMenu *menu = [[DOPDropDownMenu alloc] initWithOrigin:CGPointMake((screenSize.width - kDropDownWidth) / 2.f, kDropDownYPosition) andHeight:kDropDownHeight];
```
Comment (jannatulabeda) [inline FileReaderApp/ViewController.m]: You could use here self. menu = [[DOPDropDownMenu alloc] initWithOrigin:CGPointMake((screenSize.width - kDropDownWidth) / 2.f, kDropDownYPosition) andHeight:kDropDownHeight];
Why are you using extra variable ? :P

**Comment by jannatulabeda** [comment]
👍

**Pull request merged**